argusdashboard/src/AppBundle/Form/GenerateAlertType.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace AppBundle\Form;
+
+use Doctrine\ORM\EntityRepository;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Form\FormEvent;
+
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\Extension\Core\Type\DateType;
+use Symfony\Component\Form\Extension\Core\Type\IntegerType;
+
+use Symfony\Component\Validator\Constraints as Assert;
+
+class GenerateAlertType extends ConfigurationAbstractType

```

Comment:
could you try to use inheritance here in order to refactor some identical code. ?
I can see that the switch($type) is  duplicated in each Generate Alert / Weekly / Monthly Type class. 
You can have a mother class mybe abstract that can handle lot of common code.